notebooks/integrating_auth_mast_service.ipynb

```diff
@@ -0,0 +1,455 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**This notebook serves to demonstrate how to integrate the `auth.mast` service into a Django web application.  The code was adopted from the solution derived for the `jwql` web application.  Much of the code was written by Christian Mesh of the Archive Science Branch.  Thank you Christian!**\n",
+    "\n",
+    "**Please note that this notebook does not work right out of the box, and some of the cells will not run without error.  There are various tokens and metadata that must be provided, and the code within is designed for a specific project.  It may be beneficial to structure the code in a different way for different applications.  This notebook simply serves as a guide.**\n",
+    "\n",
+    "**Resources:**\n",
+    "- The `auth.mast` service can be found here: https://auth.mast.stsci.edu/\n",
+    "- The development version of the `auth.mast` service can be found here:  https://auth.mastdev.stsci.edu/\n",
+    "\n",
+    "\n",
+    "**In this bare-bones example, we assume the following directory structure for the `django` web application:**\n",
+    "\n",
+    "```\n",
+    "website/   \n",
+    "    apps/\n",
+    "        my_app/\n",
+    "            views.py\n",
+    "            urls.py\n",
+    "            templates/\n",
+    "                some_webpage.html\n",
+    "```\n",
+    "\n",
+    "**Required third-party libraries**\n",
+    "\n",
+    "- `authlib`\n",
+    "- `django`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Necessary imports\n",
+    "import requests\n",
+    "\n",
+    "from authlib.django.client import OAuth\n",
+    "from django.shortcuts import redirect, render\n",
+    "from django.urls import path"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set some necessary global parameters\n",
+    "client_id = ''  # Provided by ASB\n",
+    "client_secret = ''  # Provided by ASB\n",
```

Comment:
What does someone have to do to get the `client_id` and `client_secret` values? Are those just copied directly from the `config.json` on the VM, or are they individual-specific? (I can't remember how I got mine, unfortunately...) It might be a good idea to put a sentence or two about where to get those, either from JWQL members in the config file or by emailing ASB, as I anticipate that would be a question someone might ask.

Reply:
In the particular case of JWQL, they are indeed grabbed from the `config.json` file.  They are specific to the application, not an individual user.  But yes, I can be more clear here about this here.

FYI, I am hopefully going to be working with @cmesh-stsci soon to get a dummy project with its own `client_id` and `client_secret` that can be shared and used as an example.  If/when that happens, we could just put those tokens here.